### PR TITLE
Use DBIterator, improve tests

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: inc/user.class.php [hunk 1]
```diff
@@ -4302,13 +4323,22 @@ public function showForgetPassword($email) {
...
-                                   AND (`glpi_users`.`end_date` IS NULL
-                                        OR `glpi_users`.`end_date` > NOW())")) {
+      $condition = [
+         'glpi_users.is_active'  => 1,
+         'glpi_users.is_deleted' => 0,
+         'OR' => [
+            ['glpi_users.begin_date' => null],
+            ['glpi_users.begin_date' => ['<', new QueryExpression('NOW()')]]
+         ],
+         'AND' => [
```
Comment (orthagh) [inline inc/user.class.php]: this parent AND seems superflu

File: ajax/autocompletion.php [hunk 2]
```diff
@@ -55,49 +55,57 @@
...
 if (isset($_GET['entity_restrict']) && $_GET['entity_restrict']>=0) {
    if ($item->isEntityAssign()) {
-      $entity = " AND `entities_id` = '".$_GET['entity_restrict']."' ";
+      $entity['entities_id'] = $_GET['entity_restrict'];
    }
 }
 
 if (isset($_GET['user_restrict']) && $_GET['user_restrict']>0) {
-   $entity = " AND `users_id` = '".$_GET['user_restrict']."' ";
+   $entity['users_id'] = $_GET['user_restrict'];
```
[the remark this answers is not in the document]
Comment (trasher) [reply ajax/autocompletion.php]: Indeed... But I do not know if that was really intended

File: inc/user.class.php (hunk 1 above)
Comment (trasher) [reply inc/user.class.php]: But this is not... Without this one, we have 2 "OR" key in the array, and only the second one stays.

File: ajax/autocompletion.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (orthagh) [reply ajax/autocompletion.php]: just a hint, seems not an issue imo

**Pull request merged**